### DISCUSSION TITLE
Wrong validation of category fields

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#2918] [Feature Requests]
I'm not sure this is a bug. When creating a category field, the validation of its name checks against previously [created fields of the entire site](https://github.com/ExpressionEngine/ExpressionEngine/blob/34d567e4022988a4b25018e5b1605222fa23f526/system/ee/ExpressionEngine/Model/Category/CategoryField.php#L48).

```php
'field_name' => 'required|alphaDash|unique[site_id]|validateNameIsNotReserved|maxLength[32]',
```

However, I think it should be against the same category group.

```php
'field_name' => 'required|alphaDash|unique[group_id]|validateNameIsNotReserved|maxLength[32]',
```

I tested and this change works. Am I right? Thank you!

I'm running eemcs 6.0.6.

**Comment by intoeetive** [comment]
It looks like that's safe to change from data point of view, however needs to be tested code-wise whether it does not break any of queries.

I also wonder if we need extra validation here so that category custom field cannot be same as entry custom field, so there will be no conflicts
